pkg/cluster/hive_test.go

```diff
@@ -272,3 +272,61 @@ func createManagerForTests(t *testing.T, existingNamespaceName string) *manager
 	}
 	return m
 }
+
+func TestHiveDeleteResources(t *testing.T) {
+	for _, tt := range []struct {
+		testName           string
+		namespace          string
+		clusterManagerMock func(mockCtrl *gomock.Controller, namespaceName string) *mock_hive.MockClusterManager
+		wantErr            string
+	}{
+		{
+			testName: "doesn't return error if cluster manager is nil",
+		},
+		{
+			testName:  "deletes namespace if it exists",
+			namespace: "existing-namespace",
+			clusterManagerMock: func(mockCtrl *gomock.Controller, namespaceName string) *mock_hive.MockClusterManager {
+				mockClusterManager := mock_hive.NewMockClusterManager(mockCtrl)
+				mockClusterManager.EXPECT().Delete(gomock.Any(), namespaceName).Return(nil)
+				return mockClusterManager
+			},
+		},
+		{
+			testName: "doesn't attempt to delete namespace if it doesn't exist",
+			clusterManagerMock: func(mockCtrl *gomock.Controller, namespaceName string) *mock_hive.MockClusterManager {
+				mockClusterManager := mock_hive.NewMockClusterManager(mockCtrl)
+				mockClusterManager.EXPECT().Delete(gomock.Any(), gomock.Any()).Times(0)
```

Comment:
Same as in [the other PR](https://github.com/Azure/ARO-RP/pull/2308#discussion_r938803574): need to assert args

```suggestion
				mockClusterManager.EXPECT().Delete(gomock.Any(), namespaceName).Times(0)
```

Reply:
I completely agree with the other case where we should be passing the NS - but I think this particular one ("don't expect `Delete` to ever be called") seems reasonable to me.